### PR TITLE
Fix StructReader outputPositions copied from nested streams

### PR DESCRIPTION
**Pull request by bhhari**
Same fix as in #13655 applied to struct reader.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/StructSelectiveStreamReader.java [hunk 1]
```diff
@@ -270,7 +270,12 @@ public int read(int offset, int[] positions, int positionCount)
                 }
                 else {
                     readNestedStreams(offset, positions, positionCount);
-                    outputPositions = nestedOutputPositions;
+                    if (nestedOutputPositions != null) {
```
Comment (mbasmanova) [inline presto-orc/src/main/java/com/facebook/presto/orc/reader/StructSelectiveStreamReader.java]: @bhhari 

- `else` block seems unnecessary
- `nestedOutputPositions != null` check raises a question of why would nestedOutputPositions be null; checking nestedOutputPositionCount might be clearer

```
if (nestedOutputPositionCount  > 0) {
    outputPositions = initializeOutputPositions(outputPositions, nestedOutputPositions, positionCount);
}
```

Comment (bhhari) [reply presto-orc/src/main/java/com/facebook/presto/orc/reader/StructSelectiveStreamReader.java]: yes this is much clear

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/StructSelectiveStreamReader.java [hunk 2]
```diff
@@ -561,7 +563,7 @@ private void compactValues(int[] positions, int positionCount, boolean compactNu
         outputPositionCount = positionCount;
     }
 
-    private BlockLease newLease(Block block, BlockLease...fieldBlockLeases)
+    private BlockLease newLease(Block block, BlockLease... fieldBlockLeases)
```
Comment (mbasmanova) [inline presto-orc/src/main/java/com/facebook/presto/orc/reader/StructSelectiveStreamReader.java]: @bhhari unrelated change?

Comment (bhhari) [reply presto-orc/src/main/java/com/facebook/presto/orc/reader/StructSelectiveStreamReader.java]: removed
